### DISCUSSION TITLE
Im new and dont understand the coding

### DISCUSSION THREAD

**Discussion by Simracer07** [azzajess/USB-Shifter-to-Thrustmaster-Wheelbase#7] [Q&A] [answered]
So i really like the idea but i don’t understand anything:(
Could you make some kinda of list what i have to buy and do( i dont have the shifter yet and searched for something like this).

is the arduino something i can reuse or will it stay in the case? And do you have some more pictures ?

i hope you can help me

**Comment by azzajess** [comment]
There are two methods of installation that will depend on the type of shifter and what kind of method your able to do. 
USB Shield Method - https://github.com/azzajess/USB-Shifter-to-Thrustmaster-Wheelbase/blob/main/USBShield.md
Hardwire Method - https://github.com/azzajess/USB-Shifter-to-Thrustmaster-Wheelbase/blob/main/ShifterDirectWire.md

These pages contain what's required depending on each method.

The reason we have two is that some of the USB shifters encountered may not work with the USB Shield method. 

So, best thing to do is first order the USB Shifter you want from aliexpress/ebay/anywhere else.

But regardless of which method you definitely need these items.
- Arduino Pro Mini 328 - 3.3V/8MHz (It has to be 3v3 volts to power off of the wheelbase easily)
- USB to TTL Serial that supports 3v3 programing for flashing ino 
- Mini-Din/PS2 plug (for plugging into the wheelbase.) 
- USB Shifter of course

Your more than likely going to be required to solder and manually wire things up. Or maybe you can utilise an old PS2/Mini-Din keyboard or cable to use?

Again, the USB Shield method is more designed for those that want to be able to switch between PC and console with their shifter, since we are retaining the USB cable on shifter. If we do the hard wire method, its a more permanent solution.
Only thing is we had a report of the USB shield method not working, and its because the shifter is different to mine. Anyway, no way of knowing exactly unless you get the shifter first.

Rather than explaining things poorly on my part, the best course of action is to watch some YouTube vids on how to actually flash arduino pro mini and using a usb ttl serial. There the videos will also show you how to flash ino file, do some debug etc etc. I wont be able to do the explaining justice.


> is the arduino something i can reuse or will it stay in the case? And do you have some more pictures ?

By reuse, you mean use for another project? Yeah, as long as your dont need the shifter to work with console anymore. The arduino is required to be connected if your using your shifter with t300

Here are some extra images

https://github.com/azzajess/USB-Shifter-to-Thrustmaster-Wheelbase/tree/main/Images

I hope that this has made things a little more clear, and I haven't just confused you more.

**Reply by Simracer07** [reply]
And i want to do the hardwire since i can connect it to the base 


And i can connect the shifter directky to the base so i can use it on pc and ps4

**Reply by Simracer07** [reply]
And do you have a pic of all the stuff stotred and combined ? 
In the pictures are also usb shield pics

**Reply by azzajess** [reply]
I don't because this project started with the USB shield method. I created the hardwire method after.

**Comment by Simracer07** [comment]
Thank you can you use the 7+r shifter too?

**Reply by azzajess** [reply]
Yeah, shouldn't be a problem. Especially if your doing the hard wire method, then should be relatively simple to follow once you got your pinout correct

**Reply by Simracer07** [reply]
> Yeah, shouldn't be a problem. Especially if your doing the hard wire method, then should be relatively simple to follow once you got your pinout correct

Thats where i need help i dont understant how i can find out wich pin does wich gear

**Reply by azzajess** [reply, marked as answer]
You have to open up the shifter and probe the board with a multi meter on continuity mode. The board should look similar to the image pinout in the hardwire GitHub page. You can use either probe from behind it use some male to male DuPont connectors to make it easier to probe. Just stick them in the hole. 

Generally you find the common ground first. Trying to make this as simple as possible, look at the image on the hardwire shifter and match the position of the board to the image. Use that as starting reference. Then put one probe from the multimeter to the ground pin like the image and the other probe in the other ground. If you hear a solid beep you found ground
Then with one probe still connected to ground, connect the other probe on the next pin down then press each switch until you hear a beep. Keep going until you mapped out each pin with a gear.

Your ground may or may not be on a different pin. If you can't find the common ground you may have to find it from the micro switches, but don't worry about that just yet, test the above first
